overlord/snapstate/conflict.go

```diff
@@ -90,6 +90,34 @@ func affectedSnaps(t *state.Task) ([]string, error) {
 	return nil, nil
 }
 
+// CheckChangeKindExclusiveConflict checks for conflicts with a new change which
+// must be run when no other changes are running.
+func CheckChangeKindExclusiveConflict(st *state.State, newChangeKind string) error {
```

Comment:
The name sounds a bit unclear, we want to conflict with any non-ready change regardless of its kind,  maybe `CheckChangeConflictAnyRunningChange`? The comment should say that newChangeKind is only for the error message and doesn't affect conflict logic, e.g.
```
CheckChangeConflictAnyRunningChange checks that there are no unready changes and if so, returns ChangeConflictError. newChangeKind is only used for the error message.
```